include/seqan3/alphabet/aminoacid/aa20.hpp

```diff
@@ -0,0 +1,425 @@
+        J = L,
+        O = L,
+        U = C,
+        X = S,
+        Z = E,
+        TERMINATOR = W,
+        UNKNOWN = S
+    };
+
+    //!\brief Value to char conversion table.
+    static constexpr char_type value_to_char[value_size]
+    {
+        'A',
+        'C',
+        'D',
+        'E',
+        'F',
+        'G',
+        'H',
+        'I',
+        'K',
+        'L',
+        'M',
+        'N',
+        'P',
+        'Q',
+        'R',
+        'S',
+        'T',
+        'V',
+        'W',
+        'Y',
+    };
+
+    //!\brief Char to value conversion table.
+    static constexpr std::array<internal_type, 256> char_to_value
+    {
+        [] () constexpr
+        {
+            using in_t = internal_type;
+            std::array<in_t, 256> ret{};
+
+            // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
+            for (auto & c : ret)
+                c = in_t::UNKNOWN;
+
+            ret['A'] = in_t::A; ret['a'] = in_t::A;
+            ret['B'] = in_t::D; ret['b'] = in_t::D; // Convert b (either D/N) to D, since D occurs more frequently.
+            ret['C'] = in_t::C; ret['c'] = in_t::C;
+            ret['D'] = in_t::D; ret['d'] = in_t::D;
+            ret['E'] = in_t::E; ret['e'] = in_t::E;
+            ret['F'] = in_t::F; ret['f'] = in_t::F;
+            ret['G'] = in_t::G; ret['g'] = in_t::G;
+            ret['H'] = in_t::H; ret['h'] = in_t::H;
+            ret['I'] = in_t::I; ret['i'] = in_t::I;
+            ret['J'] = in_t::L; ret['j'] = in_t::L; // Convert j (either I/L) to L, since L occurs more frequently.
+            ret['K'] = in_t::K; ret['k'] = in_t::K;
+            ret['L'] = in_t::L; ret['l'] = in_t::L;
+            ret['M'] = in_t::M; ret['m'] = in_t::M;
+            ret['N'] = in_t::N; ret['n'] = in_t::N;
+            ret['O'] = in_t::L; ret['o'] = in_t::L; // Convert Pyrrolysine to lysine.
+            ret['P'] = in_t::P; ret['p'] = in_t::P;
+            ret['Q'] = in_t::Q; ret['q'] = in_t::Q;
+            ret['R'] = in_t::R; ret['r'] = in_t::R;
+            ret['S'] = in_t::S; ret['s'] = in_t::S;
+            ret['T'] = in_t::T; ret['t'] = in_t::T;
+            ret['U'] = in_t::C; ret['u'] = in_t::C; // Convert Selenocysteine to cysteine.
+            ret['V'] = in_t::V; ret['v'] = in_t::V;
+            ret['W'] = in_t::W; ret['w'] = in_t::W;
+            ret['X'] = in_t::S; ret['x'] = in_t::S; // Convert unknown amino acids to serine.
+            ret['Y'] = in_t::Y; ret['y'] = in_t::Y;
+            ret['Z'] = in_t::E; ret['z'] = in_t::E; // Convert z (either E/Q) to E, since E occurs more frequently.
+            ret['*'] = in_t::W; // The most common stop codon is UGA. This is most similar to a Tryptophan.
+            return ret;
+        }()
+    };
+
+public:
+    //!\privatesection
+    //!\brief The data member.
+    internal_type _value;
+    //!\publicsection
+};
+
+constexpr aa20 aa20::A{internal_type::A};
+constexpr aa20 aa20::B{internal_type::D};
```

Comment:
You could do the mapping here, too, i.g. `constexpr aa20 aa20::B{aa20::D};`, because this is visible from the documentation and it makes it eaiser to distinguish remapped characters from fully supported ones.